src/metabase/lib/equality.cljc

```diff
@@ -104,106 +109,365 @@
   (mbql.u.match/match-one a-ref
     [:field _opts (_id :guard pos-int?)]))
 
-(defn find-closest-matching-ref
-  "Find the ref that most closely matches `a-ref` from a sequence of `refs`. This is meant to power things
-  like [[metabase.lib.breakout/breakoutable-columns]] which are supposed to include `:breakout-position` for columns
-  that are already present as a breakout; sometimes the column in the breakout does not exactly match what MLv2 would
-  have generated. So try to figure out which column it is referring to.
+(mu/defn resolve-field-id :- lib.metadata/ColumnMetadata
```

Comment:
I wouldn't be able to guess this is the namespace to look for this function. I would look for it in some metadata namespace.

Reply:
Yeah, I agree that this location sucks (see the docstring).

It used to be in `lib/field` and only used from there. Now it's used from `lib/field` and `lib/equality`. `lib/field` requires `lib/equality`. `resolve-field-id` requires `lib.card/saved-question-metadata`, and of course everyone, including `lib.card`, requires `lib.metadata.calculation`.

So I can see these options:
1. Move `resolve-field-id`, `lib.card/saved-question-metadata`, and the latter's private helpers to somewhere shared, probably `lib.metadata.calculation`.
2. Move `resolve-field-id` to a new namespace, perhaps `metabase.lib.metadata.resolve` (`resolvers`?)
3. Leave it where it is.

If we're unhappy with where it is in this PR then I think option 2 is the way to go. Thoughts?